velox/exec/tests/LimitTest.cpp

```diff
@@ -63,3 +63,43 @@ TEST_F(LimitTest, basic) {
 
   assertQueryReturnsEmptyResult(makePlan(12'345, 10));
 }
+
+TEST_F(LimitTest, basicMap) {
+  vector_size_t batchSize = 1'000;
+  std::vector<RowVectorPtr> vectors;
+  for (int32_t i = 0; i < 3; ++i) {
+    vectors.push_back(makeRowVector(
+        {makeFlatVector<int64_t>(batchSize, [](auto row) { return row; }),
+         makeMapVector<int64_t, int64_t>(
+             batchSize,
+             [](auto row) { return 2; },
+             [](auto row) { return row % 2; },
+             [](auto row) { return row % 2 + 1; })}));
+  }
+  createDuckDbTable(vectors);
+  auto makePlan = [&](int32_t offset, int32_t limit) {
+    return PlanBuilder().values(vectors).limit(offset, limit, false).planNode();
+  };
+
+  assertQuery(makePlan(0, 10), "SELECT * FROM tmp LIMIT 10");
```

Comment:
Maybe put these `assertQuery`s in a function and reuse them with the `basic` test?